ow_faults/src/FaultInjector.cpp

```diff
@@ -26,6 +32,20 @@ void FaultInjector::faultsConfigCb(ow_faults::FaultsConfig& faults, uint32_t lev
   m_faults = faults;
 }
 
+void FaultInjector::setSystemFaultMessage(ow_faults::SystemFaults& msg) {
+  // for now only arm execution errors
+  msg.header.stamp = ros::Time::now();
+  msg.header.frame_id = '/world';
+  msg.value = 4;
```

Comment:
Since you call setSystemFaultMessage from multiple faults, this value can be affected more than once. Shouldn't it be msg.value |= 0x4; ? And should 0x4 be defined with a group of other bitmasks with descriptive names?

Reply:
Hey @mogumbo , here I was following the convention set by JPL. In their messages, the name in question is always 'value'. AFAIK Anna mentioned that these were all int values, which changed depending on the type of error that occured. Last we spoke, I brought up what happens if more than 1 type of error occurs and she said they hadn't thought through that use case yet. I'll ask her but given her OOO status right now, I'm not sure when she'll be back so I left it for now. Let me know if you want to still move away from their convention or stay for now.

Reply:
To clarify, once we can determine how they want to update the System Fault message, I would adjust this function to determine what the value should be.

Reply:
It appears system fault status values are all power of two like a regular bitmask, but they aren't being used like a bitmask. Arm fault values are not defined like a bitmask at all, but they should be in case more than one fault occurs simultaneously. To me this all appears to have a broken design and we should resolved it before we merge this branch.

I think the values you have commented in the .msg files should be put into enums somewhere instead of being comments. And my strong preference is to resolve the design with Anna before we call this finished.

Ideally, msg.value would use the type of some typedef'ed enum, but that is probably too inconvenient or impossible to do in a ROS message.

Reply:
Yes, I think there just hasn't been any work so far that's yet justified that discussion so now is the time to figure it out. I can send an email to Anna to discuss this and hopefully, we can get a response back from her before EOY.

Reply:
Indeed, Lanssie is following what Anna defined, as I'm working with the same message files, which are in the owlat_msgs package.   It's not clear to me whether the arm faults are disjoint (in which case the increasing numbers are fine) or whether they can occur together (in which case a bitmask makes more sense).  I agree we should get this straightened out with Anna when she is available -- emailing her now is a good idea.   

I'd be fine with merging this to enable continued development on our end, if the possible redefinition of the fault values won't create a headache later.

Reply:
Sent and cc'd everyone on my email to Anna. Also updated the code to have Enums, so I think we have resovled what we can. With respect to bitmask, we can see what Anna says about how they want to resolve multiple faults in the messages.